lib/cf_app_discovery/client.rb

```diff
@@ -11,21 +11,42 @@ def apps
       paginator = Paginator.new do |next_url|
         get(next_url || "/v2/apps")
       end
-      paginator.flat_map do |page|
+
+      resources = paginator.flat_map do |page|
         page.fetch(:resources)
       end
+
+      resources.each do |resource|
+        resource[:route] = get_first_route(resource)
+      end
+      resources
     end
 
     def app(app_guid)
-      get("/v2/apps/#{app_guid}")
+      resource = get("/v2/apps/#{app_guid}")
+      resource[:route] = get_first_route(resource)
+      resource
     end
 
     def service_binding(service_binding_id)
       get("/v2/service_bindings/#{service_binding_id}")
     end
 
+    def routes(service_binding_id)
+      get("/v2/apps/#{service_binding_id}/routes")
+    end
+
   private
 
+    def get_first_route(resource)
+      routes_data = routes(resource.dig(:metadata, :guid)).fetch(:resources)
+      if routes_data.first.nil?
+        resource.dig :entity, :name
```

Comment:
what's the reason for using `:name` if there are no routes?  Surely if there are no routes, then the app is simply unreachable, so it's an error condition?

If we're deliberately making a target that will fail, we should at least document that we're doing so.

Reply:
I had a look into this, when the target updater runs, it will also pick up apps that don't have any routes as it checks against all the apps in PaaS.
Perhaps it would be better to set the route to `nil` or ''?

Reply:
I think there's a legitimate error case where either:

 - a user binds an app that has no routes
 - a user removes the last route from an app that is already bound

I just don't know exactly how we handle this gracefully.  There are options:

 - generate a known-bad target config, so that prometheus shows some failed scrapes
 - don't generate a config (or generate an "inactive" config) - but how does the user know they broke something?

The current behaviour is (by accident rather than design) to generate a config that will fail to scrape and so prometheus will show that the app is down (via the `up` metric).  Maybe that's worth preserving? But if it is, we should document it.

Reply:
ok, lets document the behaviour in the README and make users aware in other docs as well.